### PR TITLE
Updated client.yaml with strategy matrix, deleted client.test.yaml

### PR DESCRIPTION
**Pull request by KarishmaGhiya**

### CODE REVIEW

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -30,17 +33,40 @@ jobs:
...
+      Windows:
+        os.name: 'Windows'
+        os.vmImage: 'vs2017-win2016'
+         
+  pool:
+    vmImage: $(os.vmImage)
+
+  steps:
+    - script: |
+        mvn jetty:run-forked -Dorg.slf4j.simpleLogger.defaultLogLevel=error -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn --batch-mode -f pom.client.xml
```
Comment (scbedd) [inline .azure-pipelines/client.yml]: This looks like it's hardcoded to pom.client.xml. Is this correct?

Comment (scbedd) [reply .azure-pipelines/client.yml]: Asking because on line 64, the maven pom file is referred to by $(pomFile).

Comment (weshaggard) [reply .azure-pipelines/client.yml]: Yes we should use `$(pomFile)` here.

Comment (KarishmaGhiya) [reply .azure-pipelines/client.yml]: Just updated the commit to reflect the changes corresponding to the updated value of $pomFile

File: .azure-pipelines/client.yml [hunk 2]
```diff
@@ -30,17 +33,40 @@ jobs:
...
+      Windows:
+        os.name: 'Windows'
+        os.vmImage: 'vs2017-win2016'
+         
+  pool:
+    vmImage: $(os.vmImage)
+
+  steps:
+    - script: |
+        mvn jetty:run-forked -Dorg.slf4j.simpleLogger.defaultLogLevel=error -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn --batch-mode -f $pomFile
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: You are going to need to change this to `$(pomFile)` otherwise DevOps will not correctly map it.

**Pull request merged**